syntax = "proto3";

package xray.app.stats.command;
option go_package = "xray.website/core/app/stats/command";

message Stat {
  string name = 1;
  int64 value = 2;
}

message GetStatsRequest {
  string name = 1;
  bool reset = 2;
}

message GetStatsResponse {
  Stat stat = 1;
}

message QueryStatsRequest {
  string pattern = 1;
  bool reset = 2;
}

message QueryStatsResponse {
  repeated Stat stat = 1;
}

message SysStatsRequest {}

message SysStatsResponse {
  int64 NumGoroutine = 1;
  int64 NumGC = 2;
  int64 Alloc = 3;
  int64 TotalAlloc = 4;
  int64 Sys = 5;
  int64 Mallocs = 6;
  int64 Frees = 7;
  int64 LiveObjects = 8;
  int64 PausedTotalNs = 9;
  int32 Uptime = 10;
}

service StatsService {
  rpc GetStats(GetStatsRequest) returns (GetStatsResponse) {}
  rpc QueryStats(QueryStatsRequest) returns (QueryStatsResponse) {}
  rpc GetSysStats(SysStatsRequest) returns (SysStatsResponse) {}
}
